python/core/auto_generated/qgsdataprovider.sip.in

```diff
@@ -122,6 +122,15 @@ Returns the provider's temporal capabilities.
 
 This may be ``None``, depending on the data provider.
 
+.. versionadded:: 3.14
+%End
+
+    virtual const QgsDataProviderTemporalCapabilities *temporalCapabilities() const;
```

Comment:
Don't forget to update the existing raster and vector data providers to implement the const version too

Reply:
and SIP_SKIP, no?

Reply:
Yes, correct.

Thinking about this - it might be nicer to make the const version non virtual:

    const QgsDataProviderTemporalCapabilities* temporalCapabilities() const { return const_cast< QgsDataProvider* >( this )->temporalCapabilities(); }

?

Reply:
yes, very nice and smart! 
But, a little cons, there will be no visibility of this const method in the derived class... easier to implement  but maybe future issue ?

Reply:
Yeah, I'm not sure. Something feels "off" about the double virtual method though...

Reply:
Finally, as the compiler looks to no like very much this elegant way, I kept the first choice.